src/app/components/HighlightedText/index.tsx

```diff
@@ -67,29 +71,51 @@ interface HighlightedTextProps {
  */
 export const HighlightedText: FC<HighlightedTextProps> = ({
   text,
-  pattern,
-  part,
+  pattern = NoHighlights,
+  partsToHighlight,
   options = defaultHighlight,
 }) => {
   const { sx = defaultHighlightStyle, findOptions = {} } = options
 
   // Have we been told what to highlight exactly? If not, look for the pattern
-  const task = part ?? findTextMatch(text, [pattern], findOptions)
+  const matches = partsToHighlight ?? findTextMatches(text, pattern, findOptions)
 
   if (text === undefined) return undefined // Nothing to display
-  if (task === NO_MATCH) return <span>{text}</span> // We don't have to highlight anything
-
-  const beginning = text.substring(0, task.startPos)
-  const focus = text.substring(task.startPos, task.endPos)
-  const end = text.substring(task.endPos)
-
-  return (
-    <span style={{ textWrap: 'nowrap' }}>
-      {beginning}
-      <Box component="mark" sx={sx}>
-        {focus}
-      </Box>
-      {end}
-    </span>
+  if (!matches.length) return text // We don't have to highlight anything
+
+  const sortedMatches = matches.sort((a, b) =>
+    a.startPos > b.startPos ? 1 : a.startPos < b.startPos ? -1 : 0,
   )
+
+  const pieces: ReactNode[] = []
+  let processedChars = 0
+  let processedTasks = 0
+
+  while (processedChars < text.length) {
+    // Do we still have tasks?
+    if (processedTasks < sortedMatches.length) {
+      // Yes, there are more tasks
+      const task = sortedMatches[processedTasks]
+      if (task.startPos < processedChars) {
+        // This task with collude
```

Comment:
Still more "task" to rename

Reply:
Also, I don't know what "This task with collude" means.
with => would? collude => collide?